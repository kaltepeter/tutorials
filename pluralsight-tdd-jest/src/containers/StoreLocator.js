import React, {Component} from 'react';
import Header from '../components/Header';
import Button from '../components/Button';
import Map from '../components/Map';

class StoreLocator extends Component {
    render() {
        return (
            <div>
                <Header />
                <Button location="Portland" />
                <Button location="Astoria" />
                <Button location="" />
                <Map />
            </div>
        );
    }
}

export default StoreLocator;