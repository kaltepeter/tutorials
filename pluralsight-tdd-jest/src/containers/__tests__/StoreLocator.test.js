import React from 'react';
import {shallow} from 'enzyme';
import StoreLocator from '../StoreLocator';

describe('StoreLocator', () => {
  let mountedStoreLocator;

  beforeEach(() => {
    mountedStoreLocator = shallow(<StoreLocator />);
  });

  it('renders without crashing', () => {
    let mountedStoreLocator = shallow(<StoreLocator />);
  });

  it('renders the header', () => {
    const header = mountedStoreLocator.find('Header');
    expect(header.length).toBe(1);
  });

  it('renders two buttons', () => {
    const buttons = mountedStoreLocator.find('Button');
    expect(buttons.length).toBe(3);
  });

  it('renders the map', () => {
    const map = mountedStoreLocator.find('Map');
    expect(map.length).toBe(1);
  });
  
});

describe('ChooseMap', () => {
  it('updates this.state.currentMap when the location is passed to it', () => {
    const mountedStoreLocator = shallow(<StoreLocator />);
    const mockEvent = {target: {value: 'testland'}};
    mountedStoreLocator.instance().chooseMap(mockEvent);
    expect(mountedStoreLocator.instance().state.currentMap).toBe('testland.png');
  });
});
