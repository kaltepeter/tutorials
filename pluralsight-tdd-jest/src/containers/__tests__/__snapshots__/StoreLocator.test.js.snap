// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocator renders correctly 1`] = `
<div>
  <div
    className="Header"
  >
    <img
      alt="Wired Brain"
      src="images/wired-brain-coffee-logo.png"
    />
  </div>
  <div
    className="buttons"
  />
  <div
    className="MapBox"
  >
    <img
      src="images/none.png"
    />
  </div>
</div>
`;
